cleanup.sh

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+
+# This script does not need to be changed for certbots DNS challenge.
+# Please see the .dedynauth file for authentication information.
+
+(
+
+DEDYNAUTH=$(pwd)/.dedynauth
+
+if [ ! -f .dedynauth ]; then
+    >&2 echo "File $DEDYNAUTH not found. Please place .dedynauth file in appropriate location."
+    exit 1
+fi
+
+source .dedynauth
+
+if [ -z "$DEDYN_TOKEN" ]; then
+    >&2 echo "Variable \$DEDYN_TOKEN not found. Please set DEDYN_TOKEN=(your dedyn.io token) to your dedyn.io access token in $DEDYNAUTH, e.g."
+    >&2 echo ""
+    >&2 echo "DEDYN_TOKEN=d41d8cd98f00b204e9800998ecf8427e"
+    exit 2
+fi
+
+if [ -z "$DEDYN_NAME" ]; then
+    >&2 echo "Variable \$DEDYN_NAME not found. Please set DEDYN_NAME=(your dedyn.io name) to your dedyn.io name in $DEDYNAUTH, e.g."
+    >&2 echo ""
+    >&2 echo "DEDYN_NAME=foobar.dedyn.io"
+    exit 3
+fi
+
+if [ -z "$CERTBOT_DOMAIN" ]; then
```

Comment:
True! Actually, since we can `PUT` an empty records array to delete an rrset, we can reuse pretty much all of the code and just focus on manipulating `acme_records`. If you agree with https://github.com/desec-utils/certbot-hook/pull/13#discussion_r383116473 it's just a matter of testing for `$CERTBOT_AUTH_OUTPUT` to decide whether to add to or subtract from `acme_records` before republishing.

Reply:
Yes, that's great! I'm really impressive that you're willing to improve this script so much. Thanks a lot!

Reply:
Looks like the cleanup verification loop did end up making the `timeout_ttl` unnecessary; dropping that commit. Pushing what I have now, but I'll work on another commit tonight that merges the two scripts.